internal/v4/api.go

```diff
@@ -104,7 +106,14 @@ func New(store *charmstore.Store, config charmstore.ServerParams) *Handler {
 			// endpoints not yet implemented:
 			// "color": router.SingleIncludeHandler(h.metaColor),
 		},
-	}, h.resolveURL, h.authorizeEntity, h.entityExists)
+		Restricted: map[string]router.RestrictedIdHandler{
```

Comment:
I see what you mean, but kind of don't like the "/restricted" endpoint, as it is no more special as any other restricted endpoint, imho. You just need to be a member of a certain group to have access to it.